applications/RANSApplication/custom_processes/rans_apply_exact_nodal_periodic_condition_process.cpp

```diff
@@ -0,0 +1,321 @@
+        reorder_process.Execute();
+    }
+}
+
+std::string RansApplyExactNodalPeriodicConditionProcess::Info() const
+{
+    return std::string("RansApplyExactNodalPeriodicConditionProcess");
+}
+
+void RansApplyExactNodalPeriodicConditionProcess::PrintInfo(std::ostream& rOStream) const
+{
+    rOStream << this->Info();
+}
+
+void RansApplyExactNodalPeriodicConditionProcess::PrintData(std::ostream& rOStream) const
+{
+}
+
+void RansApplyExactNodalPeriodicConditionProcess::CreatePeriodicConditions()
+{
+    KRATOS_TRY
+
+    auto& r_base_model_part = mrModel.GetModelPart(mBaseModelPartName);
+    int condition_id = r_base_model_part.NumberOfConditions();
+    Properties::Pointer p_properties = r_base_model_part.CreateNewProperties(
+        r_base_model_part.NumberOfProperties() + 1);
+
+    auto& r_master_model_part_nodes = mrModel.GetModelPart(mMasterModelPartName).Nodes();
+    auto& r_slave_model_part_nodes = mrModel.GetModelPart(mSlaveModelPartName).Nodes();
+
+    KRATOS_ERROR_IF(r_master_model_part_nodes.size() != r_slave_model_part_nodes.size())
+        << this->Info() << " failed. Master and slave model part nodes mismatch.\n";
+
+    const int number_of_nodes = r_master_model_part_nodes.size();
+    const double eps = std::numeric_limits<double>::epsilon();
+    const array_1d<double, 3> translation = mTranslationDirection * mTranslationMagnitude;
+
+    if (mRotationAngle > eps && mTranslationMagnitude > eps) {
+#pragma omp parallel for shared(condition_id)
+        for (int i_node = 0; i_node < number_of_nodes; ++i_node) {
+            auto& r_master_node = *(r_master_model_part_nodes.begin() + i_node);
+            const array_1d<double, 3>& master_initial_position =
+                r_master_node.GetInitialPosition().Coordinates();
+            const array_1d<double, 3>& master_final_position =
+                CalculateRotatedPosition(master_initial_position + translation);
+            int& r_master_patch_index = r_master_node.FastGetSolutionStepValue(PATCH_INDEX);
+            for (int j_node = 0; j_node < number_of_nodes; ++j_node) {
+                const auto& r_slave_node = *(r_slave_model_part_nodes.begin() + j_node);
+                if (norm_2(master_final_position -
+                           r_slave_node.GetInitialPosition().Coordinates()) < mTolerance) {
+                    r_master_patch_index = r_slave_node.Id();
+                    break;
+                }
+            }
+
+            KRATOS_ERROR_IF(r_master_patch_index == 0)
+                << "Slave node is not found in " << mSlaveModelPartName
+                << " near " << master_final_position << " with tolerance "
+                << mTolerance << " for master node id=" << r_master_node.Id()
+                << " at " << r_master_node.GetInitialPosition().Coordinates()
+                << " in " << mMasterModelPartName << ".\n";
+
+            if (static_cast<std::size_t>(r_master_patch_index) != r_master_node.Id()) {
```

Comment:
I think you don't need the cast in here.

Reply:
PATCH_INDEX is of type int and Ids are of type unsigned int ryt? So there will be a compiler warning if I don't cst it properly ryt?

Reply:
You're right. Resolving.